imap/jmap_contact.c

```diff
@@ -1782,6 +1753,18 @@ static json_t *jmap_contact_from_vcard(const char *userid,
         }
     }
 
+    if (defaultEmailIndex < 0)
+        defaultEmailIndex = 0;
+    int i, size = json_array_size(emails);
+    for (i = 0; i < size; i++) {
+        json_t *item = json_array_get(emails, i);
+        json_object_set_new(item, "isDefault",
+                            i == defaultEmailIndex ? json_true() : json_false());
```

Comment:
There always is exactly one defaultEmailIndex, so couldn't this loop be replaced with

`json_object_set_new(json_array_get(emails, defaultEmailIndex), "isDefault", json_true())`

and set "isDefault" false earlier when the email JSON object is created?

(even if there is not a single email, it is safe to call array_get and set_new on NULL values)

Reply:
Yes, we can make that change.  This code is a direct copy of the original.

Reply:
Yeah, let's use the opportunity to clean this up.